clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/security/DcosAccountCredentials.groovy

```diff
@@ -0,0 +1,157 @@
+package com.netflix.spinnaker.clouddriver.dcos.security
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.netflix.spinnaker.clouddriver.dcos.cache.Keys
+import com.netflix.spinnaker.clouddriver.dcos.deploy.util.id.MarathonPathId
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials
+
+import static com.netflix.spinnaker.clouddriver.dcos.DcosConfigurationProperties.LinkedDockerRegistryConfiguration
+
+class DcosAccountCredentials implements AccountCredentials<DcosCredentialMap> {
+  private static final String CLOUD_PROVIDER = Keys.PROVIDER
+
+  final String name
+  final String account
+  final String environment
+  final String accountType
+  final List<LinkedDockerRegistryConfiguration> dockerRegistries
+  final List<String> requiredGroupMembership
+  final List<DcosRegion> regions
+  // Not really a fan of creating this just for use within deck, but it works for now
+  final List<DcosClusterInfo> dcosClusters
+  final DcosCredentialMap dcosClusterCredentials
+
+  DcosAccountCredentials(String account,
```

Comment:
Make this private so you can only construct via the builder (which does validation)